### ISSUE TITLE
Error: unable to retrieve cluster costs over time with certain GCP nodes

### ISSUE THREAD

**Issue by dwbrown2** [kubecost/cost-model#247]
Running cost-analyzer-1.48.3 and k8s 1.14.7
Receving the same error as Anup reported before: unable to retrieve cluster costs 
Issue started after upgrading. Downgrading didnt work.
Im using the standalone prometheus instance
Kubecost metrics are OK in the Prometheus targets page

```
I1129 19:42:10.562635       1 gcpprovider.go:850] Warning: no pricing data found for us-central1,n2standard,preemptible: &{map[beta.kubernetes.io/arch:amd64 beta.kubernetes.io/fluentd-ds-ready:true beta.kubernetes.io/instance-type:n2-standard-4 beta.kubernetes.io/os:linux cloud.google.com/gke-nodepool:pool-1 cloud.google.com/gke-os-distribution:cos cloud.google.com/gke-preemptible:true failure-domain.beta.kubernetes.io/region:us-central1 failure-domain.beta.kubernetes.io/zone:us-central1-a kubernetes.io/arch:amd64 kubernetes.io/hostname:gke-clvr-prod-pool-1-c3aca41a-lzs2 kubernetes.io/os:linux providerID:gce://clvr-k8s/us-central1-a/gke-clvr-prod-pool-1-c3aca41a-lzs2]}
I1129 19:42:10.562712       1 costmodel.go:993] Error getting node. Error: Warning: no pricing data found for &{map[beta.kubernetes.io/arch:amd64 beta.kubernetes.io/fluentd-ds-ready:true beta.kubernetes.io/instance-type:n2-standard-4 beta.kubernetes.io/os:linux cloud.google.com/gke-nodepool:pool-1 cloud.google.com/gke-os-distribution:cos cloud.google.com/gke-preemptible:true failure-domain.beta.kubernetes.io/region:us-central1 failure-domain.beta.kubernetes.io/zone:us-central1-a kubernetes.io/arch:amd64 kubernetes.io/hostname:gke-clvr-prod-pool-1-c3aca41a-lzs2 kubernetes.io/os:linux providerID:gce://clvr-k8s/us-central1-a/gke-clvr-prod-pool-1-c3aca41a-lzs2]}
I1129 19:42:10.571893       1 costmodel.go:2677] [Profiler] Processing Query Data took 9.429455ms
I1129 19:42:10.572164       1 main.go:276] aggregation: warm cache: 2d
I1129 19:42:21.523983       1 gcpprovider.go:850] Warning: no pricing data found for us-central1,n2standard,preemptible: &{map[beta.kubernetes.io/arch:amd64 beta.kubernetes.io/fluentd-ds-ready:true beta.kubernetes.io/instance-type:n2-standard-4 beta.kubernetes.io/os:linux cloud.google.com/gke-nodepool:pool-1 cloud.google.com/gke-os-distribution:cos cloud.google.com/gke-preemptible:true failure-domain.beta.kubernetes.io/region:us-central1 failure-domain.beta.kubernetes.io/zone:us-central1-a kubernetes.io/arch:amd64 kubernetes.io/hostname:gke-clvr-prod-pool-1-c3aca41a-lzs2 kubernetes.io/os:linux providerID:gce://clvr-k8s/us-central1-a/gke-clvr-prod-pool-1-c3aca41a-lzs2]}
I1129 19:42:21.524068       1 costmodel.go:993] Error getting node. Error: Warning: no pricing data found for &{map[beta.kubernetes.io/arch:amd64 beta.kubernetes.io/fluentd-ds-ready:true beta.kubernetes.io/instance-type:n2-standard-4 beta.kubernetes.io/os:linux cloud.google.com/gke-nodepool:pool-1 cloud.google.com/gke-os-distribution:cos cloud.google.com/gke-preemptible:true failure-domain.beta.kubernetes.io/region:us-central1 failure-domain.beta.kubernetes.io/zone:us-central1-a kubernetes.io/arch:amd64 kubernetes.io/hostname:gke-clvr-prod-pool-1-c3aca41a-lzs2 kubernetes.io/os:linux providerID:gce://clvr-k8s/us-central1-a/gke-clvr-prod-pool-1-c3aca41a-lzs2]}
I1129 19:43:21.545288       1 gcpprovider.go:850] Warning: no pricing data found for us-central1,n2standard,preemptible: &{map[beta.kubernetes.io/arch:amd64 beta.kubernetes.io/fluentd-ds-ready:true beta.kubernetes.io/instance-type:n2-standard-4 beta.kubernetes.io/os:linux cloud.google.com/gke-nodepool:pool-1 cloud.google.com/gke-os-distribution:cos cloud.google.com/gke-preemptible:true failure-domain.beta.kubernetes.io/region:us-central1 failure-domain.beta.kubernetes.io/zone:us-central1-a kubernetes.io/arch:amd64 kubernetes.io/hostname:gke-clvr-prod-pool-1-c3aca41a-lzs2 kubernetes.io/os:linux providerID:gce://clvr-k8s/us-central1-a/gke-clvr-prod-pool-1-c3aca41a-lzs2]}
I1129 19:43:21.545349       1 costmodel.go:993] Error getting node. Error: Warning: no pricing data found for &{map[beta.kubernetes.io/arch:amd64 beta.kubernetes.io/fluentd-ds-ready:true beta.kubernetes.io/instance-type:n2-standard-4 beta.kubernetes.io/os:linux cloud.google.com/gke-nodepool:pool-1 cloud.google.com/gke-os-distribution:cos cloud.google.com/gke-preemptible:true failure-domain.beta.kubernetes.io/region:us-central1 failure-domain.beta.kubernetes.io/zone:us-central1-a kubernetes.io/arch:amd64 kubernetes.io/hostname:gke-clvr-prod-pool-1-c3aca41a-lzs2 kubernetes.io/os:linux providerID:gce://clvr-k8s/us-central1-a/gke-clvr-prod-pool-1-c3aca41a-lzs2]}
```

**Comment by cristian04** [comment]
More info:

I've reinstalled kubecost in the same cluster (with pre emp instances) 
Using default values.yaml

Here are my findings:

Kubecost was running fine for around 2 minutes. Then I started receiving this:

```I1201 23:18:15.503409       1 gcpprovider.go:650] Returned Page: us-central1,n1standard,preemptible,gpu : &{Cost: VCPU: VCPUCost:0.006655 RAM: RAMBytes: RAMCost:0.0008920000000000001 Storage: StorageCost: UsesBaseCPUPrice:false BaseCPUPrice: BaseRAMPrice: BaseGPUPrice: UsageType:preemptible GPU: GPUName: GPUCost:}
I1201 23:18:15.503448       1 gcpprovider.go:650] Returned Page: us-central1,pdstandard : <nil>
I1201 23:18:15.503455       1 gcpprovider.go:650] Returned Page: us-central1,n1standard,preemptible : &{Cost: VCPU: VCPUCost:0.006655 RAM: RAMBytes: RAMCost:0.0008920000000000001 Storage: StorageCost: UsesBaseCPUPrice:false BaseCPUPrice: BaseRAMPrice: BaseGPUPrice: UsageType:preemptible GPU: GPUName: GPUCost:}
E1201 23:18:15.507131       1 main.go:214] external costs: error: bigquery: dialing: google: error getting credentials using GOOGLE_APPLICATION_CREDENTIALS environment variable: open /var/configs/key.json: no such file or directory
I1201 23:18:15.705899       1 gcpprovider.go:850] Warning: no pricing data found for us-central1,n2standard,preemptible: &{map[beta.kubernetes.io/arch:amd64 beta.kubernetes.io/fluentd-ds-ready:true beta.kubernetes.io/instance-type:n2-standard-4 beta.kubernetes.io/os:linux cloud.google.com/gke-nodepool:pool-1 cloud.google.com/gke-os-distribution:cos cloud.google.com/gke-preemptible:true failure-domain.beta.kubernetes.io/region:us-central1 failure-domain.beta.kubernetes.io/zone:us-central1-a kubernetes.io/arch:amd64 kubernetes.io/hostname:gke-clvr-prod-pool-1-c3aca41a-lzs2 kubernetes.io/os:linux providerID:gce://clvr-k8s/us-central1-a/gke-clvr-prod-pool-1-c3aca41a-lzs2]}
I1201 23:18:15.705962       1 costmodel.go:993] Error getting node. Error: Warning: no pricing data found for &{map[beta.kubernetes.io/arch:amd64 beta.kubernetes.io/fluentd-ds-ready:true beta.kubernetes.io/instance-type:n2-standard-4 beta.kubernetes.io/os:linux cloud.google.com/gke-nodepool:pool-1 cloud.google.com/gke-os-distribution:cos cloud.google.com/gke-preemptible:true failure-domain.beta.kubernetes.io/region:us-central1 failure-domain.beta.kubernetes.io/zone:us-central1-a kubernetes.io/arch:amd64 kubernetes.io/hostname:gke-clvr-prod-pool-1-c3aca41a-lzs2 kubernetes.io/os:linux providerID:gce://clvr-k8s/us-central1-a/gke-clvr-prod-pool-1-c3aca41a-lzs2]}
I1201 23:18:15.712290       1 costmodel.go:2677] [Profiler] Processing Query Data took 6.530891ms
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1a086fe]
goroutine 246 [running]:
main.getPriceVectors(0x23fed20, 0xc0001c0070, 0xc000b86e00, 0x0, 0x0, 0x3fd3333333333333, 0x3ff0000000000000, 0x0, 0x0, 0x1e20140, ...)
	/app/kubecost-cost-model/aggregation.go:409 +0x3e
main.mergeVectors(0x23fed20, 0xc0001c0070, 0xc000b86e00, 0xc000891a00, 0x0, 0x0, 0x3fd3333333333333, 0x3ff0000000000000)
	/app/kubecost-cost-model/aggregation.go:398 +0x54f
main.aggregateDatum(0x23fed20, 0xc0001c0070, 0xc000b25140, 0xc000b86e00, 0x2048649, 0x9, 0x3386c48, 0x0, 0x0, 0x0, ...)
	/app/kubecost-cost-model/aggregation.go:380 +0xfb
main.AggregateCostData(0xc000b25110, 0x2048649, 0x9, 0x3386c48, 0x0, 0x0, 0x23fed20, 0xc0001c0070, 0xc000795c68, 0xc0008f16e0)
	/app/kubecost-cost-model/aggregation.go:242 +0xad8
main.ComputeAggregateCostModel(0x2397940, 0xc0003ff700, 0x204080d, 0x2, 0x2040809, 0x2, 0x2048649, 0x9, 0x3386c48, 0x0, ...)
	/app/kubecost-cost-model/aggregation.go:824 +0x1121
main.warmAggregateCostModelCache.func2(0x437b01, 0x2048649, 0x9, 0x3386c48, 0x0, 0x0, 0x0, 0x0, 0xc0002a06c0, 0x0, ...)
	/app/kubecost-cost-model/main.go:273 +0x198
created by main.warmAggregateCostModelCache
	/app/kubecost-cost-model/main.go:267 +0x115
```

```
I1201 23:21:40.898844       1 main.go:148] Looking up service  in namespace  in zone
I1201 23:21:40.913180       1 main.go:167] Checked prometheus endpoint: http://kubecost-prometheus-server.kubecost
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1a086fe]
goroutine 236 [running]:
main.getPriceVectors(0x23fed20, 0xc0001a6070, 0xc0012d7a40, 0x0, 0x0, 0x3fd3333333333333, 0x3ff0000000000000, 0x0, 0x0, 0x1e20140, ...)
	/app/kubecost-cost-model/aggregation.go:409 +0x3e
main.mergeVectors(0x23fed20, 0xc0001a6070, 0xc0012d7a40, 0xc000596000, 0x0, 0x0, 0x3fd3333333333333, 0x3ff0000000000000)
	/app/kubecost-cost-model/aggregation.go:398 +0x54f
main.aggregateDatum(0x23fed20, 0xc0001a6070, 0xc001274bd0, 0xc0012d7a40, 0x2048649, 0x9, 0x3386c48, 0x0, 0x0, 0x0, ...)
	/app/kubecost-cost-model/aggregation.go:380 +0xfb
main.AggregateCostData(0xc001274ba0, 0x2048649, 0x9, 0x3386c48, 0x0, 0x0, 0x23fed20, 0xc0001a6070, 0xc0011ddc68, 0xc0006ec140)
	/app/kubecost-cost-model/aggregation.go:242 +0xad8
main.ComputeAggregateCostModel(0x2397940, 0xc0004a1bc0, 0x2040805, 0x2, 0x2040809, 0x2, 0x2048649, 0x9, 0x3386c48, 0x0, ...)
	/app/kubecost-cost-model/aggregation.go:824 +0x1121
main.warmAggregateCostModelCache.func1(0x437b01, 0x2048649, 0x9, 0x3386c48, 0x0, 0x0, 0x0, 0x0, 0xc0003c40f0, 0x0, ...)
	/app/kubecost-cost-model/main.go:258 +0x198
created by main.warmAggregateCostModelCache
	/app/kubecost-cost-model/main.go:252 +0xaa
```

```
NAME                                                      READY   STATUS             RESTARTS   AGE
cost-analyzer-checks-1575241800-rpgpv                     0/1     Completed          0          14m
cost-analyzer-checks-1575242400-c97n6                     0/1     Completed          0          4m49s
kubecost-cost-analyzer-8665cbb99c-nvqct                   0/3     CrashLoopBackOff   4          3m35s
kubecost-grafana-844d4b9844-wvbcs                         3/3     Running            0          15m
kubecost-prometheus-alertmanager-85bbbd6b7b-wl9dt         2/2     Running            0          15m
kubecost-prometheus-kube-state-metrics-857c5d4b4f-5qzqr   1/1     Running            0          15m
kubecost-prometheus-node-exporter-7ffn2                   1/1     Running            0          15m
kubecost-prometheus-node-exporter-hvjls                   1/1     Running            0          15m
kubecost-prometheus-pushgateway-7689458dc9-f7n8f          1/1     Running            0          15m
kubecost-prometheus-server-7b8b759d74-99cbl               2/2     Running            0          15m
```

Summary:
- Issue seems to be located on kubecost-cost-analyzer
- `I1201 23:21:40.898844       1 main.go:148] Looking up service  in namespace  in zone`

**Comment by cristian04** [comment]
Update: `The CrashLoopBackOff` was resolved by upgrading to v1.49.0

Still having issues with `warning Kubecost cost-model metrics available in Prometheus`

**Comment by dwbrown2** [comment]
Cristian and I are discussing on Slack...

**Comment by cristian04** [comment]
Verified as fixed on v1.49.0

Great job guys!

**Comment by AjayTripathy** [comment]
Happy to help!
